### PR TITLE
test(e2e): test traffic split in permissive mode

### PR DESCRIPTION
**Pull request by michelleN**
Signed-off-by: Michelle Noorali <minooral@microsoft.com>

**Description**: adds e2e test for #2527 - traffic split in permissive mode

**Testing done**:

**Affected area**:
| Functional Area            |     |
| -------------------------- | --- |
| New Functionality          | [ ] |
| CI System                  | [ ] |
| CLI Tool                   | [ ] |
| Certificate Management     | [ ] |
| Control Plane              | [ ] |
| Demo                       | [ ] |
| Documentation              | [ ] |
| Egress                     | [ ] |
| Ingress                    | [ ] |
| Install                    | [ ] |
| Networking                 | [ ] |
| Observability              | [ ] |
| Performance                | [ ] |
| SMI Policy                 | [ ] |
| Security                   | [ ] |
| Sidecar Injection          | [ ] |
| Tests                      | [X ] |
| Upgrade                    | [ ] |
| Other                      | [ ] |

Please answer the following questions with yes/no.

1. Does this change contain code from or inspired by another project? No
    -   Did you notify the maintainers and provide attribution? No

2. Is this a breaking change? No

### CODE REVIEW

File: tests/e2e/e2e_permissive_trafficsplit_test.go [hunk 1]
```diff
@@ -0,0 +1,299 @@
+package e2e
```
Comment (shashankram) [inline tests/e2e/e2e_permissive_trafficsplit_test.go:1]: I was wondering if it made sense to extend the existing e2e_trafficsplit_test.go that adds an additional variable in the top-level context for SMI vs permissive mode: https://github.com/openservicemesh/osm/blob/main/tests/e2e/e2e_trafficsplit_test.go#L24-L32. That way, we would only need to set the mode and everything in the test would remain the same, which makes it easier to maintain the test.

Comment (michelleN) [reply tests/e2e/e2e_permissive_trafficsplit_test.go:1]: yea I agree. That's a great idea.

Comment (shashankram) [reply tests/e2e/e2e_permissive_trafficsplit_test.go:1]: I extended the existing traffic split test to permissive mode in https://github.com/openservicemesh/osm/pull/4074/commits

**Comment by shashankram** [comment]
Traffic split test in permissive mode has been added as a part of https://github.com/openservicemesh/osm/pull/4074/commits: https://github.com/openservicemesh/osm/blob/main/tests/e2e/e2e_trafficsplit_test.go#L35-L38. Closing this since it's not required.

**Pull request closed without merging**